### PR TITLE
Add @infix annotation

### LINKED ISSUE #5937: Possible syntax for infix operators

### PR DESCRIPTION
**Pull request by odersky**
An elaboration of the proposal made by @sjrd in #5937. `@alpha` will come in a separate PR.

### CODE REVIEW

**Comment by odersky** [comment]
Current status: The new annotations are defined and explained, but the compiler does not interpret them yet.

File: docs/docs/reference/changed-features/operators.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+s1 union s2         // OK
+s1.union(s2)        // also OK
+
+s1.difference(s2)   // OK
+s1 `difference` s2  // OK
+s1 difference s2    // gives a deprecation warning
+
+s1 * s2             // OK
+s1.*(s2)            // also OK, but unusual
```
Comment (sjrd) [inline docs/docs/reference/changed-features/operators.md]: The two "also OK" cases (`s1.union(s2)` and `s1.*(s2)`) kind of go against the motivation of achieving consistency. For consistency to be really applied, they should give a deprecation warning.

Comment (odersky) [reply docs/docs/reference/changed-features/operators.md]: Yes, it's a tradeoff. I believe though that declaring something @infix would be a strong enough social cue that people would follow it without the compiler having to enforce it. Always allowing method call syntax has the advantage that we can normalize programs. We can say
`x union y` desugars to `x.union(y)`. If `x.union(y)` was not legal that explanation would be not available.

Comment (sjrd) [reply docs/docs/reference/changed-features/operators.md]: Meh. Yes, OK.

**Comment by odersky** [comment]
The @infix part is done now.

**Pull request merged**